src/lib/util/extract-user.ts

```diff
@@ -1,7 +1,7 @@
 import { IAuthRequest, IUser } from '../server-impl';
 
 export function extractUsernameFromUser(user: IUser): string {
-    return user ? user.email || user.username : 'unknown';
```

Comment:
Isn't this the same? Can you unit-test where this fails?

Reply:
Oh, I think I know what you said in the description, something like:
```typescript
user: {}
```
would lead to undefined. Still, a unit test would be good in this case

Reply:
Added some tests in https://github.com/Unleash/unleash/pull/5061/commits/19804ba03b483465c8e0cb6dd4ac7a8212142772.
Before this change, the 3rd test "Should return "unknown" if neither email nor username exists" would fail. We would also have a type error on the method, since it is specifying that we should return `string` instead of `string | undefined`.